layers/meta-balena-jetson/recipes-bsp/uefi/edk2-firmware-tegra/0001-Switch-boot-chains.patch

```diff
@@ -0,0 +1,213 @@
+From 93149fb12aae634aaf1be77920f565528c2ab534 Mon Sep 17 00:00:00 2001
+From: Alexandru Costache <alexandru@balena.io>
+Date: Mon, 13 May 2024 13:48:08 +0300
+Subject: [PATCH] Switch boot chains when bootcount limit is reached
+
+In this commit we rely on the hostapp-update-hooks to save
+the target firmware boot chain (A/B - 0/1) in /mnt/boot/bootchain.env.
+
+When the max bootcount value is reached, we thus know which is the
+updated target slot and can thus:
+ - set the facllback (old) bootchain as next boot target and reboot
+ - detect that we falled back to the old chain in the previous boot,
+   and continue to boot it in altboot mode. After this point the OS
+   will run the rollback hooks.
+
+The hostapp-update hooks always overite the bootchain file, and remove it
```

Comment:
*and can remove it completely upon rolling back, if necessary. Leaving it in place does not hurt either, and allows for inspection in case of any inconsistencies.